### DISCUSSION TITLE
Auto pip install -r requirements.txt

### DISCUSSION THREAD

**Discussion by KendallDoesCoding** [ThioJoe/YT-Spammer-Purge#213] [Ideas]
We can make it so that, when users run the script in python. It will automatically run pip install -r requirements.txt

**Comment by KendallDoesCoding** [comment]
@ThioJoe Is this possible?

**Comment by compsup** [comment]
Do you mean the .exe or .py script? Either way should be able to install os (`import os`) and do a `os.system('pip3 install -r requirements.py')` at the top of the file. Maybe make it ask before doing so.

**Reply by KendallDoesCoding** [reply]
py script, instead of you having to type pip3 install -r requirements.py I thought we could make it so that any time a user runs the program, it first auto executes pip3 install -r requirements.py but idk if its possible or no.

**Comment by Kurdakov** [comment]
as you might observe - almost nobody uses this capability while it is available. The reason, I think, is that is considered bad practice: someone who runs python script would most probably know what pip install -r requirements.txt means, also user has context - if python3 is default one should run pip install -r requirements.txt or if python2.7 is also installed and is default (as is on Macs) in this case 
pip3  install -r requirements.txt should be run. 

pip3 though should always work, but see above 

edit: 
consider scenario: someone runs YT Spammer Purge in virtual environment ( I do this, for example ), then consider 
that by mistake someone did not activate virtual environment and packages were installed to global python ( which was not intended). so we end up with unintended behavior, which is exactly a bad practice.

so in fact - automatic installation adds more confusion, than benefits, except cases where it is necessary and there are restrictions, when it is known - automatic installation won't lead to problems ( such as in embedded systems without virtual environments)
